visualization/include/visualization/xml_node.hpp

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <vector>
+#include <utility>
+#include <string>
+
+namespace visualization::render {
+
+    class XMLNode {
+    public:
+        class Builder;
+
+        const std::vector<std::pair<std::string, std::string>> get_attributes() const;
+        const std::string get_tag() const;
```

Comment:
`const std::vector<std::pair<std::string, std::string>>&` and `const std::string&` to avoid useless copy